src/operators/switch.ml

```diff
@@ -195,9 +200,6 @@ object (self)
             | None -> ()
           end
     in
-      (* #select is called only when selected=None, and the cache is cleared
-       * as soon as the new selection is set. *)
-      assert (selected=None || cached_selected=None) ;
```

Comment:
Why is this going away?

Reply:
Good catch thanks for asking. I had an `assert` error when testing the code. I couldn't find the source of the error, unfortunately. but I was also able to confirm that this assertion is not enforced in the code. I'll revisit this before merging.